src/usbhostmanager.cpp

```diff
@@ -0,0 +1,216 @@
+        uint8_t const* p_desc   = tu_desc_next(desc_cfg);
+
+        // parse each interfaces
+        while( p_desc < desc_end )
+        {
+            uint8_t assoc_itf_count = 1;
+            // Class will always starts with Interface Association (if any) and then Interface descriptor
+            if ( TUSB_DESC_INTERFACE_ASSOCIATION == tu_desc_type(p_desc) )
+            {
+                tusb_desc_interface_assoc_t const * desc_iad = (tusb_desc_interface_assoc_t const *) p_desc;
+                assoc_itf_count = desc_iad->bInterfaceCount;
+
+                p_desc = tu_desc_next(p_desc); // next to Interface
+            }
+
+            // must be interface from now
+            if( TUSB_DESC_INTERFACE != tu_desc_type(p_desc) ) return;
+            tusb_desc_interface_t const* desc_itf = (tusb_desc_interface_t const*) p_desc;
+
+            // only open and listen to HID endpoint IN
+            if (desc_itf->bInterfaceClass == TUSB_CLASS_HID)
+            {
+                _intf_num = desc_itf->bInterfaceNumber;
+                break; // we got the interface number
+            }
+            
+            // next Interface or IAD descriptor
+            uint16_t const drv_len = count_interface_total_len(desc_itf, assoc_itf_count, (uint16_t) (desc_end-p_desc));
+            p_desc += drv_len;
+        }
+    } // This block can be removed once TinyUSB library incorporates HID_REQ_CONTROL_GET_REPORT callback
+
+    USBHostManager::getInstance().hid_mount_cb(dev_addr, instance, desc_report, desc_len);
+    if ( !tuh_hid_receive_report(dev_addr, instance) ) {
+        // Error: cannot request report
+    }
+}
+
+/// Invoked when device is unmounted (bus reset/unplugged)
+void tuh_hid_umount_cb(uint8_t daddr, uint8_t instance)
+{
+    USBHostManager::getInstance().hid_umount_cb(daddr, instance);
+}
+
+// Invoked when received report from device via interrupt endpoint
+void tuh_hid_report_received_cb(uint8_t dev_addr, uint8_t instance, uint8_t const* report, uint16_t len)
+{
+    USBHostManager::getInstance().hid_report_received_cb(dev_addr, instance, report, len);
+
+    if ( !tuh_hid_receive_report(dev_addr, instance) ) {
+        //Error: cannot request report
+    }
+}
+
+// On IN/OUT/FEATURE set report callback
+void tuh_hid_set_report_complete_cb(uint8_t dev_addr, uint8_t instance, uint8_t report_id, uint8_t report_type, uint16_t len) {
+    if ( len != 0 )
+        USBHostManager::getInstance().hid_set_report_complete_cb(dev_addr, instance, report_id, report_type, len);
+}
+
+
+// GET REPORT FEATURE
+void tuh_hid_get_report_complete_cb(uint8_t dev_addr, uint8_t instance, uint8_t report_id, uint8_t report_type, uint16_t len) {
+    if ( len != 0 )
+        USBHostManager::getInstance().hid_get_report_complete_cb(dev_addr, instance, report_id, report_type, len);
+}
+
+// Request for HID_REQ_CONTROL_GET_REPORT missing from TinyUSB
+static void get_report_complete(tuh_xfer_t* xfer)
+{
+  if (tuh_hid_get_report_complete_cb)
```

Comment:
This is a direct pull from the TinyUSB PR I submitted at the same time: https://github.com/hathach/tinyusb/pull/2223

I wanted to keep this code as-close-as-possible to the PR I submitted, since this code really should be in the TinyUSB library.